classroom.rb

```diff
@@ -0,0 +1,60 @@
+# Represents a classroom with attributes such as label and students
+class Classroom
+  attr_accessor :label, :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    students << student
+    student.classroom = self
+  end
+end
+
+# Represents a student with attributes name and classroom.
+class Student
+  attr_accessor :name, :classroom
+
+  def initialize(name)
+    @name = name
+    @classroom = nil
+  end
+end
+
+# Represents a book with attributes such as title, author, and rentals.
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
+
+require 'date'
+# Represents a rental with attributes such as date, book, and person.
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+
+    book.rentals << self
+    person.rentals << self
+  end
+end
+
+# Represents a person with attributes such as name and rentals.
+class Person
+  attr_accessor :name, :rentals
+
+  def initialize(name)
+    @name = name
+    @rentals = []
+  end
```

Comment:
- [ ] Please create a `rentals` variable in the `Person` class that was created in the `person.rb` file and assign it the value of an empty array. After creating the variable, assign the getters and setters method (`attr_accessor`) to it for easy access. Remember that the `Book` and `Person` classes have a **_many-to-many_** relationship through the `Rental` class. This can be successfully implemented with the creation of the `@rentals` array and the `add_rental` methods in both the `Book` and `Person` classes. This would enable their instances to keep records of all the rentals that belong to that instance. 👍 😎 
- [ ] Finally, I would advise you to create the `add_rental` method in the `Person` class to successfully implement the relationship between both classes. The `add_rental` method should create a new instance of the `Rental` class. 💻 🖱️ 

#### Hint:

```ruby
def add_rental
  Rental.new(date, book, self)
end
```

Reply:
@Gambit142 done